### PR TITLE
inital commit for the Event Synchronization class

### PR DESCRIPTION
**Pull request by wbarfuss**
Event Synchronization (ES) class with three basic Event Synchronization matrices:
- directed
- symmetric or undirected
- antisymmetric

By that, I pretty much sticked to the original ES paper. The actual ES measure between two event series is calculated via vectorization. The unit test checks if this vectorization coincides with the straight forward loop procedure. 

Looking forward to the discussion

### CODE REVIEW

**Comment by ntfrgl** [comment]
Cool! Don't mind the failing Travis CI build, this is still the same [Conda issue](https://github.com/conda/conda/issues/1367). Some suggestions:
-  Add reference to the paper in docstrings and ``docs/source/publications.rst``, see e.g. ``climate.tsonis``.
-  Cache ``directed()`` using ``core.network.cached_const`` decorator (you could export it from ``core.__init__``).
-  ``np.arange`` -> ``xrange`` in ``directed()``.
-  No need to allocate ``DSTyx`` in ``_EventSync()``, just use reciprocal comparisons with ``DSTxy``.
-  Reuse ``np.minimum(TAU, self.taumax)`` and ``np.sum(DSTxy == 0)``.
-  Self-referential definition of ``__str__()``?
- ``np.allclose`` instead of ``(==).all()`` in ``testVectorization()``
-  "the the" in ``__init__()``, "eventsynchronization" in ``NonVecEventSync()``, "straigth" in ``testVectorization()``.

**Comment by wbarfuss** [comment]
Thanks. I tried to improve the code so far following your suggestions. I'm happy to continue to do so.
We should also think of how to connect the event synchronization class with the network class. I think this still needs to be done.

**Comment by ntfrgl** [comment]
Congratulations, you have found a bug! This is the way to call a method from a parent class, which doesn't make any sense for classes directly derived from ``object``, thus the self-referential ``__str__``. This has been fixed in daf187eb1c51acc7bdabc95000f1552ecb7ffacf. Please adapt your ``__str__`` and include an appropriate test instance in ``tests/test_generic.py``.

"evenmatrix" in ``__init__()``, "efficte Tau" in ``_EventSync()``. Otherwise, LGTM!

What should be the conceptual connection between the two classes?

**Comment by jdonges** [comment]
The idea is to create an EventSyncClimateNetwork class that inherits from both the ClimateNetwork and the EventSynchronization classes. This is very straightforward and can follow the example of the RecurrenceNetwork class that inherits from Network and RecurrencePlot without introducing much functionality itself at all.

One would like to create an event sync climate network from a ClimateData object as easily as possible. Here, I would suggest that the EventSyncClimateNetwork supports quantile thresholding to convert time series to event series data. Methodologically, this should go along the lines of Niklas Boers et al. papers on the subject.

**Comment by wbarfuss** [comment]
Something similar as in the 45b0762? Open for any suggestions :)

**Comment by ntfrgl** [comment]
- ``EventSyncClimateNetwork.__init__()`` should check for a valid ``eventsynctype`` parameter.
- (n)``ode_weight_type``
- ``winter_only`` parameter has no consequences (used only in correlation climate networks so far).
- Add ``EventSyncClimateNetwork`` to ``tests/test_generic.py``.

File: docs/source/publications.rst [hunk 1]
```diff
@@ -347,6 +348,25 @@ Time series analysis and synchronization (general)
     `doi:10.1103/PhysRevE.86.061121
     <http://dx.doi.org/10.1103/PhysRevE.86.061121>`__
 
+*Event synchronization*
+~~~~~~~~~~~~~~~~~~~~~~~
+[Quiroga2002]_, [Boers2014]_.
+
+.. [Quiroga2002] R.Q. Quiroga, T. Kreuz, P. Grassberger.
+    "Quantifying causal coupling strength: A lag-specific measure for
```
Comment (jdonges) [inline docs/source/publications.rst]: Wrong title, please check ref completely.

File: docs/source/publications.rst [hunk 2]
```diff
@@ -347,6 +348,25 @@ Time series analysis and synchronization (general)
...
+    multivariate time series related to transfer entropy".
+    In *Physical Review E*, vol. 66(4), 041904 (2002)
+    `doi:10.1103/PhysRevE.66.041904
+    <http://dx.doi.org/10.1103/PhysRevE.66.041904>`__
+
+.. [Boers2014] N. Boers, B. Bookhagen, H.M.J. Barbosa, N. Marwan, J. Kurths.
+    J.A. Marengo.
+    "Event synchronization: a simple and fast method to measure synchronicity
```
Comment (jdonges) [inline docs/source/publications.rst]: Wrong title + check reference.

**Comment by jdonges** [comment]
I suggest to rename the EventSyncClimateNetwork class to EventSynchronizationClimateNetwork for consistency with the description paper. And suggest to move the class into the subpackage climate instead of funcnet.

**Pull request merged**